server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/AddressMappingRoutesTest.java

```diff
@@ -0,0 +1,143 @@
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.restassured.RestAssured;
+import io.restassured.filter.log.LogDetail;
+import net.javacrumbs.jsonunit.core.Option;
+
+class AddressMappingRoutesTest {
+
+    private WebAdminServer webAdminServer;
+    private MemoryRecipientRewriteTable recipientRewriteTable;
+
+    @BeforeEach
+    void setUp() throws DomainListException {
+        recipientRewriteTable = new MemoryRecipientRewriteTable();
+        DNSService dnsService = mock(DNSService.class);
+        DomainList domainList = new MemoryDomainList(dnsService);
+        domainList.addDomain(Domain.of("domain.tld"));
+
+        recipientRewriteTable.setDomainList(domainList);
+
+        webAdminServer = WebAdminUtils.createWebAdminServer(
+            new AddressMappingRoutes(recipientRewriteTable))
+            .start();
+
+        RestAssured.requestSpecification = WebAdminUtils.buildRequestSpecification(webAdminServer)
+            .setBasePath(AddressMappingRoutes.BASE_PATH)
+            .log(LogDetail.METHOD)
+            .build();
+    }
+
+    @AfterEach
+    void stop() {
+        webAdminServer.destroy();
+    }
+
+    @Test
+    void addAddressMappingShouldAddMappingOnRecipientRewriteTable() throws Exception {
+        MailAddress mailAddress = new MailAddress("source@domain.tld");
+
+        recipientRewriteTable.addAddressMapping(
+            MappingSource.fromMailAddress(mailAddress), "alice@domain.tld");
+        recipientRewriteTable.addAddressMapping(
+             MappingSource.fromMailAddress(mailAddress), "bob@domain.tld");
+        recipientRewriteTable.addAddressMapping(
+             MappingSource.fromMailAddress(mailAddress), "james@domain.tld");
+
+        when()
+            .put("source@domain.tld" + "/targets/" + "user@domain.tld");
+
+        assertThat(recipientRewriteTable.getStoredMappings(MappingSource.parse("source@domain.tld")))
+            .containsAnyOf(Mapping.of("user@domain.tld"));
+    }
+
+    @Test
+    void addAddressMappingShouldReturnNotFoundWhenOneParameterIsEmpty() {
+        when()
+            .put("source@domain.tld" + "/targets/" )
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404);
+    }
+
+    @Test
+    void addAddressMappingShouldReturnNoContentWhenValidParameter() {
+        when()
+            .put("source@domain.tld" + "/targets/" + "alice@domain.tld")
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+    }
+
+    @Test
+    void addAddressMappingShouldReturnBadRequestWhenInvalidParameter() {
+        when()
+            .put("source" + "/targets/" + "alice@domain.tld")
+        .then()
+            .statusCode(HttpStatus.BAD_REQUEST_400);
+    }
+
+    @Test
+    void addAddressMappingShouldFilterOutInvalidDestinationAddress() throws RecipientRewriteTableException, AddressException {
```

Comment:
This method is quite the same as the method above addAddressMappingShouldReturnBadRequestWhenInvalidParameter(). This scenario means when the destination address is invalid, it will not be added to the recipient rewrite table. Since it is obvious, I will remove it